src/software/new_geom/util/closest_point_test.cpp

```diff
@@ -0,0 +1,27 @@
+#include "software/new_geom/util/closest_point.h"
+
+#include <gtest/gtest.h>
+
+TEST(ClosestPointTest, three_points)
+{
+    Point p = closestPointOnLine(Point(0, 0), Point(2, 0), Point(0, 2));
+    EXPECT_EQ(p, Point(1, 1));
+}
+
+TEST(ClosestPointTest, point_line1)
```

Comment:
Can these test names be more descriptive? Ideally they should state what the test is testing, eg. `point_far_from_line` or `point_close_to_line_but_not_on_line`

No need to worry about making the names too long, since they only need to be written once (it's not a variable that developers have to deal with) and makes reading and fixing tests much easier.

Reply:
Changed to be more descriptive and added a few more test cases